dds/DCPS/StatusConditionImpl.cpp

```diff
@@ -16,6 +16,8 @@ namespace DCPS {
 
 CORBA::Boolean StatusConditionImpl::get_trigger_value()
 {
+  ACE_GUARD_RETURN(ACE_Recursive_Thread_Mutex, g, lock_, false);
```

Comment:
Increasing scope of the lock could have significant performance implications with debug logging on (maybe even enough to change the system behavior with logging vs. non-logging).